DATA: users, gamePlay 데이터 테이블 정의 및 관계 설정 #7

jn33-dev opened the pull request:
---
name: Pull Request
about: 3차 스프린트 목표 스콥까지 필요한 데이터를 설계하고, typeorm으로 entity 정의
title: DATA: users, gamePlay 데이터 테이블 정의 및 관계 설정 #7
labels: 'backend', 'enhancement
assignees: 최예나

---

**작업한 내용**
- ERD 작성
[ERD](https://drawsql.app/teams/gachi-1/diagrams/playgame-v0-0-1)
- ERD에 맞춰 DB 구성
- package.json에 필요한 명령어 수정, db 관련 명령어는 레포 readme 확인
- package.json 머지 과정 중 꼬인 것 clear
- 프로젝트 설정 파일(루트 폴더에 있는 것들) prettier로 format 적용


**REVIEW POINT**
- users 폴더에서 entities 폴더를 새로 생성하고 그 안에 entity 파일을 모두 밀어 넣었습니다. 꼭 확인해주세요!
- 찾아보니까, column 이름을 원하는대로 지정할 수 있더라고요. 이 부분은 user table에서 id -> userId로 할지 같이 얘기해보아요~
- README.md에 작업루틴 작성한 것 꼭 확인해주세요!!!

jn33-dev committed:
DATA : gamePlay data 중 room entity 생성 #7

- room entity 파일 작성 (관계설정 전)
Changes to be committed:
	new file:   project/src/games/room.entity.ts

jn33-dev committed:
DATA : gamePlay data entity 생성 및 관계 설정 #7

- 테스트 전 저장
Changes to be committed:
	modified:   dataSource.ts
	modified:   src/app.module.ts
	new file:   src/games/entities/player.entity.ts
	renamed:    src/games/room.entity.ts -> src/games/entities/room.entity.ts
	new file:   src/games/entities/socketIdMap.entity.ts

jn33-dev committed:
FIX : package.json 머지 과정 중에 꼬인 것 정리

Changes to be committed:
	modified:   project/package-lock.json
	modified:   project/package.json

jn33-dev committed:
DATA: gamePlay data ERD에 맞춰 typeorm 작성 #7

Changes to be committed:
	modified:   src/app.module.ts
	modified:   src/games/entities/player.entity.ts
	modified:   src/games/entities/room.entity.ts
	modified:   src/games/entities/socketIdMap.entity.ts
	modified:   src/games/games.module.ts
	modified:   src/users/auth/auth.service.ts
	new file:   src/users/entities/token-map.entity.ts
	renamed:    src/users/user.entity.ts -> src/users/entities/user.entity.ts
	modified:   src/users/users.module.ts
	modified:   src/users/users.service.ts

jn33-dev committed:
DOCS : README에 작업 루틴 가이드 작성

Changes to be committed:
	modified:   README.md

jn33-dev committed:
DOCS : 프로젝트 설정 파일 오류 수정 및 format

Changes to be committed:
	modified:   .eslintrc.js
	modified:   .prettierrc
	modified:   README.md
	modified:   nest-cli.json
	modified:   package-lock.json
	modified:   package.json
	renamed:    dataSource.ts -> src/dataSource/dataSource.ts
	modified:   tsconfig.build.json
	modified:   webpack-hmr.config.js

jn33-dev committed:
DATA : user, tokenMap, room, player, socketIdMap entities 정의 #7

- tokenMap을 추가하면서, users/entity 폴더를 생성하고, user.entity와 tokenMap.entity를 폴더 안으로 정리
Changes to be committed:
	modified:   README.md
	renamed:    project/src/dataSource/dataSource.ts -> project/dataSource.ts
	modified:   project/package.json
	modified:   project/src/app.module.ts
	modified:   project/src/common/exceptionFilters/http-exception.filter.ts
	modified:   project/src/games/dto/create-room.request.dto.ts
	modified:   project/src/games/dto/roomInfoToMain.dto.ts
	modified:   project/src/games/games.gateway.ts
	modified:   project/src/games/games.module.ts
	renamed:    project/src/games/inGame-users.service.ts -> project/src/games/players.service.ts
	modified:   project/src/main.ts
	modified:   project/src/users/auth/kakao.serializer.ts
	modified:   project/src/users/entities/user.entity.ts
	modified:   project/src/users/users.service.ts

RESOLVE : issue #7 - user & gamePlay data

jn33-dev committed:
FIX : gitignore에 package.json 추가

Changes not staged for commit:
	modified:   project/package.json

jn33-dev committed:
DOCS : README 내용 수정

Changes to be committed:
	modified:   ../README.md
	modified:   .gitignore

dy-p-code reviewed (approved):
리드미랑 엔티티 변경된 부분 확인했습니다 :) !!! 👍

jn33-dev committed:
CHORE : re-build #7

Changes to be committed:
	modified:   dist/tsconfig.build.tsbuildinfo

jn33-dev committed:
Merge branch 'dev' of https://github.com/gachimind/backend into feature/7

jn33-dev committed:
DATA : users, playGame data 관계 수정 및 column 이름 수정 #7

- users 폴더 내에 id -> userId로 모두 수정하였으나, 혹시나 작동 안하는 것이 있다면, 확인 해보기!
Changes to be committed:
	modified:   dataSource.ts
	modified:   package.json
	modified:   src/games/entities/player.entity.ts
	modified:   src/games/entities/room.entity.ts
	modified:   src/games/entities/socketIdMap.entity.ts
	deleted:    src/users/auth/auth.service.ts
	modified:   src/users/auth/kakao.data.ts
	modified:   src/users/auth/kakao.strategy.ts
	modified:   src/users/dto/create-user.dto.ts
	modified:   src/users/entities/token-map.entity.ts
	modified:   src/users/entities/user.entity.ts
	modified:   src/users/users.service.ts

RESOLVE: issue #7

jn33-dev commented:
- OAuth 배포 후 수정한 버전을 pull 받아 merge conflict 모두 해결
- 데이터 관계 설정 변경 [ERD 참고](https://drawsql.app/teams/gachi-1/diagrams/playgame-v0-0-1)
